### ISSUE TITLE
Issue with fitting procedure for newer versions of lmfit

### ISSUE THREAD

**Issue by Lars26888** [lmfit/lmfit-py#385]
I have used lmfit to fit some of my experimental data and it worked out well for version 0.9.0. After updating to version 0.9.5, I get this error message:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Python27\lib\site-packages\spyderlib\widgets\externalshell\sitecustomize.py", line 685, in runfile
    execfile(filename, namespace)
  File "C:\Python27\lib\site-packages\spyderlib\widgets\externalshell\sitecustomize.py", line 71, in execfile
    exec(compile(scripttext, filename, 'exec'), glob, loc)
  File "C:/Users/l.breuer/Desktop/lmfit_bug.py", line 60, in <module>
    result = mod.fit(T, pars, x=x)
  File "C:\Python27\lib\site-packages\lmfit\model.py", line 541, in fit
    output.fit(data=data, weights=weights)
  File "C:\Python27\lib\site-packages\lmfit\model.py", line 747, in fit
    _ret = self.minimize(method=self.method)
  File "C:\Python27\lib\site-packages\lmfit\minimizer.py", line 1168, in minimize
    return function(**kwargs)
  File "C:\Python27\lib\site-packages\lmfit\minimizer.py", line 1008, in leastsq
    lsout = scipy_leastsq(self.__residual, vars, **lskws)
  File "C:\Python27\lib\site-packages\scipy\optimize\minpack.py", line 381, in leastsq
    gtol, maxfev, epsfcn, factor, diag)
  File "C:\Python27\lib\site-packages\lmfit\minimizer.py", line 331, in __residual
    out = _nan_policy(out, nan_policy=self.nan_policy)
  File "C:\Python27\lib\site-packages\lmfit\minimizer.py", line 1357, in _nan_policy
    raise ValueError("The input contains nan values")
ValueError: The input contains nan values

I have tried several versions of lmfit and the error only occurs for versions 0.9.4 and 0.9.5 with version 0.9.3 being the last one working fine. This is a minimal working example including the data and function which illustrates the problem:

from numpy import *
from lmfit import Model

x = array([ 0.01235521,  0.02471042,  0.03706564,  0.04942085,  0.06177606,
        0.07413127,  0.08648649,  0.0988417 ,  0.11119691,  0.12355212,
        0.13590734,  0.14826255,  0.16061776,  0.17297297,  0.18532819,
        0.1976834 ,  0.21003861,  0.22239382,  0.23474903,  0.24710425,
        0.25945946,  0.27181467,  0.28416988,  0.2965251 ,  0.30888031,
        0.32123552,  0.33359073,  0.34594595,  0.35830116,  0.37065637,
        0.38301158,  0.3953668 ,  0.40772201,  0.42007722,  0.43243243,
        0.44478764,  0.45714286,  0.46949807,  0.48185328,  0.49420849,
        0.50656371,  0.51891892,  0.53127413,  0.54362934,  0.55598456,
        0.56833977,  0.58069498,  0.59305019,  0.60540541,  0.61776062,
        0.63011583,  0.64247104,  0.65482625,  0.66718147,  0.67953668,
        0.69189189,  0.7042471 ,  0.71660232,  0.72895753,  0.74131274,
        0.75366795,  0.76602317,  0.77837838,  0.79073359,  0.8030888 ,
        0.81544402,  0.82779923,  0.84015444,  0.85250965,  0.86486486,
        0.87722008,  0.88957529,  0.9019305 ,  0.91428571,  0.92664093,
        0.93899614,  0.95135135,  0.96370656,  0.97606178,  0.98841699,
        1.0007722 ,  1.01312741,  1.02548263,  1.03783784,  1.05019305,
        1.06254826,  1.07490347,  1.08725869,  1.0996139 ,  1.11196911,
        1.12432432,  1.13667954,  1.14903475,  1.16138996,  1.17374517,
        1.18610039,  1.1984556 ,  1.21081081,  1.22316602,  1.23552124,
        1.24787645,  1.26023166,  1.27258687,  1.28494208,  1.2972973 ,
        1.30965251,  1.32200772,  1.33436293,  1.34671815,  1.35907336,
        1.37142857,  1.38378378,  1.396139  ,  1.40849421,  1.42084942])

T = [19.87254, 20.1192, 20.36409, 20.24186, 20.24186, 19.99609, 19.87254, 
     19.87254, 19.87254, 19.49916, 19.49916, 19.99609, 19.74854, 19.87254, 
     20.1192, 20.1192, 19.74854, 19.99609, 19.99609, 19.62408, 19.99609, 
     19.74854, 19.99609, 19.37378, 20.1192, 19.87254, 20.24186, 19.74854, 
     19.87254, 19.74854, 19.74854, 19.74854, 20.36409, 20.1192, 20.24186, 
     20.1192, 20.1192, 19.99609, 20.60725, 20.72819, 21.08845, 21.08845, 
     21.44499, 21.56302, 21.20771, 21.79787, 22.26287, 22.83549, 23.39886, 
     24.06315, 24.71514, 25.67113, 26.29449, 27.5096, 28.00406, 28.68515, 
     28.49184, 29.0687, 28.97319, 28.49184, 27.10911, 26.60214, 25.35531, 
     24.71514, 24.06315, 23.62168, 22.83549, 22.49304, 22.03115, 21.91472, 
     21.32656, 21.32656, 20.72819, 20.60725, 20.60725, 20.60725, 20.72819, 
     20.60725, 19.99609, 20.24186, 19.62408, 19.99609, 19.87254, 20.1192, 
     19.74854, 19.87254, 19.62408, 19.62408, 19.24795, 19.74854, 19.74854, 
     19.74854, 19.24795, 19.74854, 19.37378, 19.37378, 19.87254, 19.87254, 
     19.74854, 19.62408, 19.99609, 19.62408, 19.87254, 19.62408, 19.74854, 
     19.87254, 19.62408, 19.74854, 19.87254, 19.74854, 19.49916, 19.74854, 
     19.62408, 19.74854, 19.87254]

def f(x,y0,A,w,xc,D):
    return y0 + A * exp(-2 * (x-xc) **2 / w **2) + 1/(2 * sqrt( pi * D)) * exp(-(x-xc)**2/(4*D))
    

mod = Model(f)    
pars = mod.make_params(y0=19, A=1, w=0.3, xc=0.7, D=1)
result = mod.fit(T, pars, x=x)

**Comment by newville** [comment]
@Lars26888 Yes, you are seeing the benefits of an important feature introduced in 0.9.4 to enforce a policy of what to do when NaNs are encountered in the objective function.  

Previous versions would foolishly pass such NaNs along to the solver, which typically have no mechanism for gracefully handling NaNs.  This sometimes ended up with erroneous results, and little in the way of warning that the fit did not actually complete.

Basically, the exception is saying that when evaluating your model function, some input value was causing a NaN, such as dividing by zero.   The best action to take is to make sure your model function does not return NaNs by, for example, checking that arguments and denominators are valid.

To return to the older, fragile behavior, you can also pass `fit_kws={'nan_policy': 'propagate'}`  to `Model.fit()`.
